lib/src/Pos/Diffusion/Full/Block.hs

```diff
@@ -348,7 +362,7 @@ streamBlocks logTrace smM logic streamWindow enqueue nodeId tipHeader checkpoint
     requestBlocks :: Conc.TVar Bool -> Conc.TBQueue StreamEntry -> IO (Conc.TVar (OQ.PacketStatus ()))
     requestBlocks fallBack blockChan = do
         convMap <- enqueue (MsgRequestBlocks (S.singleton nodeId))
-                            (\_ _ -> (Conversation $ \it -> requestBlocksConversation blockChan it `finally` writeStreamEnd blockChan) :|
+                            (\_ _ -> (Conversation $ \it -> requestBlocksConversation blockChan it `onException` writeStreamEnd blockChan) :|
```

Comment:
If it's essential to use `onException` rather than `finally`, then we should put a comment explaining it. Why can `requestBatch` still use `finally`?

Reply:
`requestBatch` can get away with `finally` because the only thing it will write is `StreamEnd` for which there will always be room. I will add a comment.